### ISSUE TITLE
Investigate Servo backend

### ISSUE THREAD

**Issue by Emdek** [OtterBrowser/otter-browser#1111]
Since Gecko is currently not embedding friendly and not everybody likes Blink or WebKit, we should investigate if it is possible to create Servo based backend.

**Comment by pierreporte** [comment]
Will all the work you did for QtWebEngine, which currently uses Blink as backend, be compatible with Servo? If not, it may not be worth it now as there will be a lot more to do and Otter is not yet stable.

Apart from purely technical reasons, the supremacy of Blink and Webkit in the world of rendering engines may cause a big problem in the future (I am not a specialist, but I read that it would cause the same problems as IE6 ten years ago). For this reason, using Servo would be a good choice.

**Comment by Emdek** [comment]
@pierreporte, since the unstable part is usually something wrong in upstream libraries... ;-)
It would be best if such backend would have own maintainer or something like that.

**Comment by SJ-Diamond** [comment]
As Mozilla tries so hard to be even more Chrome than Chrome itself, i wonder if Servo will have any future at all.

I would be these days very careful with everything Mozilla related.

**Comment by zakius** [comment]
I don't know if Mozilla makes even more chrome than chrome, for now firefox is superior to anything released as stable (except dreadfully outdated opera) and I hope it won't chrome itself to the level of being useless (like chrome or chropera)

**Comment by SJ-Diamond** [comment]
You have still the benefit to bring back complex customization features with add-ons like Classic theme restorer back, but the features itself are still removed from the browser itself. Just wait until Mozilla kills of XUL and implements the more restrictive Chrome extensions model, that will be the moment where the real "fun" will start.

**Comment by Emdek** [comment]
@SJ-Diamond, AFAIK currently Mozilla doesn't plan to replace Gecko by Servo, they want to use some of its component for now only, but only future will tell what they really want to do with it.

**Comment by zakius** [comment]
AFAIK Mozilla wants to keep as much possibilities in API as possible (...?), but making it trivial to run chromium extensions in firefox (but that itself doesn't mean that firefox extensions will work in chromium as they can use extended API)
I agree that they are taking some things away (and the removal of Pocket extension cause they built in "add to pocket button" was plainly dumb)
Still it's API allows for much more than anyone, so it is superior, even if goes down with every update there's a long way down to chromium

And Servo seems to be aimed mainly at mobile devices (where it could better utilize multi-threaded optimizations and algorithms, as on desktop and not-so-mobile devices like laptops dual core processors are still pretty common, and mobiles tend to have 4+4 or more), but if it goes well porting it to desktop is likely

**Comment by s8321414** [comment]
I think it's a good idea if Servo is considered as stable :smile:

**Comment by ShalokShalom** [comment]
There is a Servo browser, which renders on OpenGL directly, without any toolkit, the whole UI written in web languages.

**Comment by mariokamenjak** [comment]
Mozilla will eventually completely remove gecko. It will be a gradual transition though.

**Comment by ShalokShalom** [comment]
Yeah, I think so too. The nub of the matter is; when becomes Servo useable?

**Comment by mariokamenjak** [comment]
Their roadmap says that major features of Servo will be in stable Firefox by the end of 2017.

**Comment by Emdek** [comment]
I'm wondering if this ticket still makes sense, I haven't heard anything new in long time and it seems that Mozilla integrated some features back into Firefox creating memory hungry beast.

**Comment by Frenzie** [comment]
They way Mozilla marketing has phrased it they've been plugging parts of Servo into Firefox. I don't know. :-)

**Comment by Emdek** [comment]
@Frenzie, yeah, according to comment above they were supposed to be already done with that integration by now...

**Comment by ShalokShalom** [comment]
Two seperate projects: Servo and Mozilla. 
As you see, Servo is its own project: https://github.com/servo/servo

Mozilla decides to integrate parts and supports the project. 
The own browser/UI is still in development.

See here the current state: https://download.servo.org/

**Comment by Emdek** [comment]
@ShalokShalom, it seems that it will take some more time before it will be production ready...

**Comment by ShalokShalom** [comment]
Thanks for the consideration
